runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/StreamingEngineClient.java

```diff
@@ -164,23 +177,27 @@ public static StreamingEngineClient create(
       JobHeader jobHeader,
       GetWorkBudget totalGetWorkBudget,
       GrpcWindmillStreamFactory streamingEngineStreamFactory,
-      WorkItemProcessor processWorkItem,
+      WorkItemScheduler processWorkItem,
       ChannelCachingStubFactory channelCachingStubFactory,
       GetWorkBudgetDistributor getWorkBudgetDistributor,
-      GrpcDispatcherClient dispatcherClient) {
-    StreamingEngineClient streamingEngineClient =
-        new StreamingEngineClient(
-            jobHeader,
-            totalGetWorkBudget,
-            new AtomicReference<>(StreamingEngineConnectionState.EMPTY),
-            streamingEngineStreamFactory,
-            processWorkItem,
-            channelCachingStubFactory,
-            getWorkBudgetDistributor,
-            dispatcherClient,
-            new Random().nextLong());
-    streamingEngineClient.start();
-    return streamingEngineClient;
+      GrpcDispatcherClient dispatcherClient,
+      Function<WindmillStream.CommitWorkStream, WorkCommitter> workCommitterFactory,
+      Consumer<List<Windmill.ComputationHeartbeatResponse>> heartbeatProcessor,
+      AtomicReference<StreamingEngineConnectionState> streamingEngineConnectionsState) {
+    // Set the connectionState to a valid state if it is null.
+    streamingEngineConnectionsState.compareAndSet(null, StreamingEngineConnectionState.EMPTY);
+    return new StreamingEngineClient(
+        jobHeader,
+        totalGetWorkBudget,
+        streamingEngineConnectionsState,
+        streamingEngineStreamFactory,
+        processWorkItem,
+        channelCachingStubFactory,
+        getWorkBudgetDistributor,
+        dispatcherClient,
+        new Random().nextLong(),
```

Comment:
nit: add param comment or use a variable to document what this is.

Reply:
done